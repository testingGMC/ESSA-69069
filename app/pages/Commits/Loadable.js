/**
 *
 * Asynchronously loads the component for Commits
 *
 */

import loadable from 'utils/loadable'
export default loadable(() => import('./index'))
