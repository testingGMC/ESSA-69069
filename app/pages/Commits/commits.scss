.commits{
    padding: 2%;
    padding-left: 5%;
    padding-right: 5%;
    .ant-card-body{
        padding-top:15px;
    }
  
    .time-line-title {
        font-size: 16px;
        color: #393c40;
        small {
            color: gray;
        }
    }
    .list-item-date {
        font-size: 14px;
        color: #393c40;
        font-style: italic;
        small {
            margin-left: 5px;
            color: gray;
        }    
    }
    .list-item-title {
        font-size: 17px;
        font-weight: bold;
        color: #393c40;

    }

    .commits-undo-icon{
        font-size: 18px;
    }
}
