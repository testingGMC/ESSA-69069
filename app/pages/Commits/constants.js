export const COMMIT_PER_PAGE = 5

export const DAY_MONTH_YEAR_DATE_FORMAT = 'DD MM YYYY'
