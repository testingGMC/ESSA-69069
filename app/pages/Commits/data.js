/* @review: remove this file if we not using it  */
export const testData = [
  {
    message:
      'DependsOnRelationship (8d70bb12-e45a-45a8-8d5a-26cb63109ff8->9dd41b37-6993-45e7-be10-ead574805dda) relationship created',
    description:
      'DependsOnRelationship relationship (8d70bb12-e45a-45a8-8d5a-26cb63109ff8->9dd41b37-6993-45e7-be10-ead574805dda) created',
    author: {
      fullName: 'karim zeheni',
      email: null,
    },
    changes: [
      {
        type: 2,
        data:
          'eyJNb2RlbFR5cGUiOiJEZXBlbmRzT25SZWxhdGlvbnNoaXAiLCJJZCI6IjhkNzBiYjEyLWU0NWEtNDVhOC04ZDVhLTI2Y2I2MzEwOWZmODlkZDQxYjM3LTY5OTMtNDVlNy1iZTEwLWVhZDU3NDgwNWRkYSIsIkJsb2JUeXBlIjoxLCJEYXRhIjoie1wiZGF0ZUVmZmVjdGl2ZVwiOlwiMjAxOS0wNi0wM1QxMzozMDoyOS44NzAzNDk1KzAwOjAwXCIsXCJ3ZWlnaHRcIjowLjAsXCJJZFwiOlwiOGQ3MGJiMTItZTQ1YS00NWE4LThkNWEtMjZjYjYzMTA5ZmY4OWRkNDFiMzctNjk5My00NWU3LWJlMTAtZWFkNTc0ODA1ZGRhXCIsXCJTb3VyY2VcIjpcIjhkNzBiYjEyLWU0NWEtNDVhOC04ZDVhLTI2Y2I2MzEwOWZmOFwiLFwiVGFyZ2V0XCI6XCI5ZGQ0MWIzNy02OTkzLTQ1ZTctYmUxMC1lYWQ1NzQ4MDVkZGFcIixcIkZyb21LZXlcIjpcInNcIixcIktleVwiOlwic3RcIixcIlRvS2V5XCI6XCJ0XCJ9IiwiUGF0Y2giOmZhbHNlfQ==',
      },
    ],
    date: '2019-06-03T13:30:29.92786+00:00',
    id: 'f53ab70a-618e-4fa7-9a86-bb77d20c5d48',
  },
  {
    message:
      'DependsOnRelationship (8d70bb12-e45a-45a8-8d5a-26cb63109ff8->9dd41b37-6993-45e7-be10-ead574805dda) relationship created',
    description:
      'DependsOnRelationship relationship (8d70bb12-e45a-45a8-8d5a-26cb63109ff8->9dd41b37-6993-45e7-be10-ead574805dda) created',
    author: {
      fullName: 'ahmed houni',
      email: null,
    },
    changes: [
      {
        type: 2,
        data:
          'eyJNb2RlbFR5cGUiOiJEZXBlbmRzT25SZWxhdGlvbnNoaXAiLCJJZCI6IjhkNzBiYjEyLWU0NWEtNDVhOC04ZDVhLTI2Y2I2MzEwOWZmODlkZDQxYjM3LTY5OTMtNDVlNy1iZTEwLWVhZDU3NDgwNWRkYSIsIkJsb2JUeXBlIjoxLCJEYXRhIjoie1wiZGF0ZUVmZmVjdGl2ZVwiOlwiMjAxOS0wNi0wM1QxMzozMDoyOS44NzAzNDk1KzAwOjAwXCIsXCJ3ZWlnaHRcIjowLjAsXCJJZFwiOlwiOGQ3MGJiMTItZTQ1YS00NWE4LThkNWEtMjZjYjYzMTA5ZmY4OWRkNDFiMzctNjk5My00NWU3LWJlMTAtZWFkNTc0ODA1ZGRhXCIsXCJTb3VyY2VcIjpcIjhkNzBiYjEyLWU0NWEtNDVhOC04ZDVhLTI2Y2I2MzEwOWZmOFwiLFwiVGFyZ2V0XCI6XCI5ZGQ0MWIzNy02OTkzLTQ1ZTctYmUxMC1lYWQ1NzQ4MDVkZGFcIixcIkZyb21LZXlcIjpcInNcIixcIktleVwiOlwic3RcIixcIlRvS2V5XCI6XCJ0XCJ9IiwiUGF0Y2giOmZhbHNlfQ==',
      },
    ],
    date: '2019-06-03T13:30:29.92786+00:00',
    id: 'f53ab70a-618e-4fa7-9a86-bb77d20c5d48',
  },
  {
    message:
      'DependsOnRelationship (8d70bb12-e45a-45a8-8d5a-26cb63109ff8->9dd41b37-6993-45e7-be10-ead574805dda) relationship created',
    description:
      'DependsOnRelationship relationship (8d70bb12-e45a-45a8-8d5a-26cb63109ff8->9dd41b37-6993-45e7-be10-ead574805dda) created',
    author: {
      fullName: 'achour mohamed',
      email: null,
    },
    changes: [
      {
        type: 2,
        data:
          'eyJNb2RlbFR5cGUiOiJEZXBlbmRzT25SZWxhdGlvbnNoaXAiLCJJZCI6IjhkNzBiYjEyLWU0NWEtNDVhOC04ZDVhLTI2Y2I2MzEwOWZmODlkZDQxYjM3LTY5OTMtNDVlNy1iZTEwLWVhZDU3NDgwNWRkYSIsIkJsb2JUeXBlIjoxLCJEYXRhIjoie1wiZGF0ZUVmZmVjdGl2ZVwiOlwiMjAxOS0wNi0wM1QxMzozMDoyOS44NzAzNDk1KzAwOjAwXCIsXCJ3ZWlnaHRcIjowLjAsXCJJZFwiOlwiOGQ3MGJiMTItZTQ1YS00NWE4LThkNWEtMjZjYjYzMTA5ZmY4OWRkNDFiMzctNjk5My00NWU3LWJlMTAtZWFkNTc0ODA1ZGRhXCIsXCJTb3VyY2VcIjpcIjhkNzBiYjEyLWU0NWEtNDVhOC04ZDVhLTI2Y2I2MzEwOWZmOFwiLFwiVGFyZ2V0XCI6XCI5ZGQ0MWIzNy02OTkzLTQ1ZTctYmUxMC1lYWQ1NzQ4MDVkZGFcIixcIkZyb21LZXlcIjpcInNcIixcIktleVwiOlwic3RcIixcIlRvS2V5XCI6XCJ0XCJ9IiwiUGF0Y2giOmZhbHNlfQ==',
      },
    ],
    date: '2019-06-04T13:30:29.92786+00:00',
    id: 'f53ab70a-618e-4fa7-9a86-bb77d20c5d48',
  },
]
