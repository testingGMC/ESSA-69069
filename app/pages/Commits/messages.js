/*
 * Commits Messages
 *
 * This contains all the text for the Commits container.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.containers.Commits'

export default defineMessages({
  commits: {
    id: `${scope}.commits`,
    defaultMessage: 'Commits',
  },
  commitsHistory: {
    id: `${scope}.commitsHistory`,
    defaultMessage: 'commits history',
  },
  helmetDescription: {
    id: `${scope}.helmetDescription`,
    defaultMessage: 'Description of Commits',
  },
  helmetTitle: {
    id: `${scope}.helmetTitle`,
    defaultMessage: 'Commits',
  },
})
