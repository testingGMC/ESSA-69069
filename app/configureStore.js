/**
 * Create the store with dynamic reducers
 */

import { createStore, applyMiddleware, compose } from 'redux'
import { routerMiddleware } from 'connected-react-router'
import createSagaMiddleware from 'redux-saga'
import { NOTIFICATION_TYPES } from 'shared/constants'
import { pushNotification } from 'shared/utils/notification'
import { errorsExtraction } from 'shared/utils/error-interceptors'
import reduxMiddlewares from './reduxMiddlewares'
import createReducer from './reducers'

export default function configureStore(initialState = {}, history) {
  let composeEnhancers = compose
  const reduxSagaMonitorOptions = {}

  // If Redux Dev Tools and Saga Dev Tools Extensions are installed, enable them
  /* istanbul ignore next */
  if (process.env.NODE_ENV !== 'production' && typeof window === 'object') {
    /* eslint-disable no-underscore-dangle */
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__)
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})

    // NOTE: Uncomment the code below to restore support for Redux Saga
    // Dev Tools once it supports redux-saga version 1.x.x
    // if (window.__SAGA_MONITOR_EXTENSION__)
    //   reduxSagaMonitorOptions = {
    //     sagaMonitor: window.__SAGA_MONITOR_EXTENSION__,
    //   };
    /* eslint-enable */
  }

  // const error400Handling = store => next => action => {
  //   if (action.e && action.type.slice(-7) === 'FAILURE') {
  //     if (action.e.response.status === 400) {
  //       const errors = errorsExtraction(action.e.response.data.model)
  //       errors.forEach(error =>
  //         pushNotification(NOTIFICATION_TYPES.error, error),
  //       )
  //       next({ ...action, errors })
  //       return
  //     }
  //   }
  //   next(action)
  // }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions)

  // Create the store with two middlewares
  // 1. sagaMiddleware: Makes redux-sagas work
  // 2. routerMiddleware: Syncs the location/URL path to the state
  const middlewares = [
    sagaMiddleware,
    routerMiddleware(history),
    ...reduxMiddlewares,
  ]

  const enhancers = [applyMiddleware(...middlewares)]

  const store = createStore(
    createReducer(),
    initialState,
    composeEnhancers(...enhancers),
  )

  // Extensions
  store.runSaga = sagaMiddleware.run
  store.injectedReducers = {} // Reducer registry
  store.injectedSagas = {} // Saga registry

  // Make reducers hot reloadable, see http://mxs.is/googmo
  /* istanbul ignore next */
  if (module.hot) {
    module.hot.accept('./reducers', () => {
      store.replaceReducer(createReducer(store.injectedReducers))
    })
  }

  return store
}
