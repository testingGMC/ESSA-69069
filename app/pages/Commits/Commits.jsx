/* eslint-disable react/no-array-index-key */
/**
 *
 * Commits
 *
 */

import React, { useEffect } from 'react'
import QueueAnim from 'rc-queue-anim'
import PropTypes from 'prop-types'
import initials from 'initials'
import { FormattedMessage, injectIntl, intlShape } from 'react-intl'
import { isEmpty } from 'lodash'
import { Helmet } from 'react-helmet'
import { Row, Col, Timeline, List, Avatar, Skeleton, Tooltip, Icon } from 'antd'
import moment from 'moment'
import ColorHash from 'color-hash'
import Card from 'shared/components/Card'
import { groupByDate, orderByFullDate } from 'shared/utils/group-by'
import { renderFullName } from 'shared/utils/user.helper'
import { isContentArchitect } from 'shared/utils/access-token'
import { isFirstIndex } from 'shared/utils/array.helper'

import { COMMIT_PER_PAGE, DAY_MONTH_YEAR_DATE_FORMAT } from './constants'
import messages from './messages'
import './commits.scss'

const Commits = ({ fetchCommits, commits, revertToACommit, intl, roles }) => {
  useEffect(() => {
    fetchCommits()
  }, [])

  const commitsData =
    commits.commits !== undefined ? groupByDate(commits.commits) : []

  const helmetMessages = {
    helmetTitle: intl.formatMessage({
      ...messages.helmetTitle,
    }),
    helmetDescription: intl.formatMessage({
      ...messages.helmetDescription,
    }),
  }

  function displayCommitIcon(dateIndex, commitIndex, item) {
    if (
      (isFirstIndex(dateIndex) && isFirstIndex(commitIndex)) ||
      !isContentArchitect(roles)
    ) {
      return null
    }
    return (
      <Icon
        className="commits-undo-icon"
        type="undo"
        onClick={() =>
          revertToACommit(
            item.id,
            `Revert back to ${item.id}`,
            `Revert back to ${item.id}`,
          )
        }
      />
    )
  }

  const colorHash = new ColorHash()
  return (
    <div className="commits">
      <Helmet>
        <title>{helmetMessages.helmetTitle}</title>
        <meta name="description" content={helmetMessages.helmetDescription} />
      </Helmet>
      <Row>
        <Col>
          <Card title={<FormattedMessage {...messages.commits} />}>
            <Timeline>
              <Skeleton
                avatar
                title={false}
                loading={isEmpty(commitsData)}
                active
              >
                <QueueAnim delay={300} className="queue-simple">
                  {Object.keys(commitsData).map((key, index) => (
                    <Timeline.Item color="#e6e6e6" key={index}>
                      <div className="time-line-title">
                        {moment(key).format('MMMM Do YYYY')}{' '}
                        <small>{moment(key).fromNow()}</small>
                      </div>

                      <List
                        pagination={{ pageSize: COMMIT_PER_PAGE }}
                        className="demo-loadmore-list"
                        loading={false}
                        itemLayout="horizontal"
                        dataSource={orderByFullDate(commitsData[key])}
                        renderItem={(item, itemIndexi) => (
                          <List.Item
                            actions={[
                              displayCommitIcon(index, itemIndexi, item),
                            ]}
                            key={itemIndexi}
                          >
                            <Skeleton
                              avatar
                              title={false}
                              loading={false}
                              active
                            >
                              <List.Item.Meta
                                avatar={
                                  <Avatar
                                    style={{
                                      backgroundColor: colorHash.hex(
                                        renderFullName(item.author.fullName),
                                      ),
                                    }}
                                  >
                                    {initials(
                                      renderFullName(item.author.fullName),
                                    )}
                                  </Avatar>
                                }
                                title={
                                  <span className="list-item-title">
                                    <Tooltip title={item.message}>
                                      <span>{item.message}</span>
                                    </Tooltip>
                                  </span>
                                }
                                description={
                                  <span className="list-item-date">
                                    {item.hoursMinute}
                                    {moment(item.fullDate).format(
                                      DAY_MONTH_YEAR_DATE_FORMAT,
                                    ) ===
                                      moment().format(
                                        DAY_MONTH_YEAR_DATE_FORMAT,
                                      ) && (
                                      <small>
                                        {moment(item.fullDate).fromNow()}
                                      </small>
                                    )}
                                  </span>
                                }
                              />
                            </Skeleton>
                          </List.Item>
                        )}
                      />
                    </Timeline.Item>
                  ))}
                </QueueAnim>
              </Skeleton>
            </Timeline>
          </Card>
        </Col>
      </Row>
    </div>
  )
}

Commits.propTypes = {
  commits: PropTypes.objectOf(PropTypes.any).isRequired,
  fetchCommits: PropTypes.func.isRequired,
  revertToACommit: PropTypes.func.isRequired,
  intl: intlShape.isRequired,
  roles: PropTypes.array.isRequired,
}

export default injectIntl(Commits)
