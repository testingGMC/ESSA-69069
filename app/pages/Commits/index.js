/**
 *
 * Commits
 *
 */

import React, { memo } from 'react'
import { connect } from 'react-redux'

import { createStructuredSelector } from 'reselect'
import { compose } from 'redux'
import injectSaga from 'utils/injectSaga'
import { DAEMON } from 'utils/constants'
import { useInjectReducer } from 'utils/injectReducer'
import makeSelectCommits from '../../shared/redux/commits/selectors'
import reducer from '../../shared/redux/commits/reducer'

import * as actions from '../../shared/redux/commits/actions'

import saga from '../../shared/redux/commits/saga'

import Commits from './Commits'

const CommitsIndex = props => {
  useInjectReducer({ key: 'commits', reducer })

  return <Commits {...props} />
}

const mapStateToProps = createStructuredSelector({
  commits: makeSelectCommits(),
})

const mapDispatchToProps = {
  ...actions,
}

const withSaga = injectSaga({ key: 'commits', saga, mode: DAEMON })

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
)

export default compose(
  withConnect,
  memo,
  withSaga,
)(CommitsIndex)
