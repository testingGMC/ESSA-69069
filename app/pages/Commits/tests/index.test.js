
import React from 'react'
import { render } from 'react-testing-library'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { browserHistory } from 'react-router-dom'

import CommitsIndex from '../index'
import Commits from '../Commits'
import { DEFAULT_LOCALE } from '../../../i18n'
import configureStore from '../../../configureStore'

const initialProps = {
  fetchCommits: jest.fn(),
  fetchLatestCommits: jest.fn(),
  revertToACommit: jest.fn(),
  commits: {}
}

describe('<CommitsIndex />', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('Expect to not log in console', () => {
    const spy = jest.spyOn(global.console, 'log')
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <CommitsIndex {...initialProps} />
        </IntlProvider>
      </Provider>
    )
    expect(spy).not.toHaveBeenCalled()
  })


  it('Expect to not log errors in console', () => {
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <CommitsIndex {...initialProps} Commits />
        </IntlProvider>
      </Provider>,
    )
  })

})

describe('<Commits/>', () => {
  let store
  beforeAll(() => {
    store = configureStore({}, browserHistory)
  })

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Commits {...initialProps} commits />
        </IntlProvider>
      </Provider>,
    )
    expect(firstChild).toMatchSnapshot()

  })

  it('should fetch the Commits on mount if the CommitsId exists', () => {
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Commits {...initialProps} commits />
        </IntlProvider>
      </Provider>,
    )
    expect(initialProps.fetchCommits).toHaveBeenCalled()
  })
})