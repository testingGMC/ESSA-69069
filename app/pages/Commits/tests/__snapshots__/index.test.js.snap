// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Commits/> should render and match the snapshot 1`] = `
<div
  class="commits"
>
  <div
    class="ant-row"
  >
    <div
      class="ant-col"
    >
      <div
        class="card"
      >
        <div
          class="ant-card ant-card-small"
        >
          <div
            class="ant-card-head"
          >
            <div
              class="ant-card-head-wrapper"
            >
              <div
                class="ant-card-head-title"
              >
                <span>
                  Commits
                </span>
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="content "
            >
              <ul
                class="ant-timeline"
              >
                <div
                  class="ant-skeleton ant-timeline-item-last ant-skeleton-with-avatar ant-skeleton-active"
                >
                  <div
                    class="ant-skeleton-header"
                  >
                    <span
                      class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
                    />
                  </div>
                  <div
                    class="ant-skeleton-content"
                  >
                    <ul
                      class="ant-skeleton-paragraph"
                    >
                      <li />
                      <li
                        style="width: 61%;"
                      />
                    </ul>
                  </div>
                </div>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
